spec/beaker/dsl/install_utils/foss_utils_spec.rb

```diff
@@ -29,66 +30,62 @@ def logger
   let(:winhost)       { make_host( 'winhost', { :platform => 'windows',
                                                 :pe_ver => '3.0',
                                                 :working_dir => '/tmp',
+                                                :type => 'foss',
                                                 :is_cygwin => true} ) }
   let(:winhost_non_cygwin) { make_host( 'winhost_non_cygwin', { :platform => 'windows',
                                                 :pe_ver => '3.0',
                                                 :working_dir => '/tmp',
+                                                :type => 'foss',
                                                 :is_cygwin => 'false' } ) }
   let(:machost)       { make_host( 'machost', { :platform => 'osx-10.9-x86_64',
                                                 :pe_ver => '3.0',
+                                                :type => 'foss',
                                                 :working_dir => '/tmp' } ) }
   let(:unixhost)      { make_host( 'unixhost', { :platform => 'linux',
                                                  :pe_ver => '3.0',
                                                  :working_dir => '/tmp',
+                                                :type => 'foss',
                                                  :dist => 'puppet-enterprise-3.1.0-rc0-230-g36c9e5c-debian-7-i386' } ) }
   let(:eoshost)       { make_host( 'eoshost', { :platform => 'eos',
                                                 :pe_ver => '3.0',
                                                 :working_dir => '/tmp',
+                                                :type => 'foss',
                                                 :dist => 'puppet-enterprise-3.7.1-rc0-78-gffc958f-eos-4-i386' } ) }
 
 
   context '#configure_foss_defaults_on' do
-    it 'uses aio paths for hosts of type aio' do
+    it 'uses aio paths for hosts with role aio' do
       hosts.each do |host|
-        host[:type] = 'aio'
+        host[:pe_ver] = nil
+        host[:version] = nil
+        host[:roles] = host[:roles] | ['aio']
       end
+      expect(subject).to receive(:add_foss_defaults_on).exactly(hosts.length).times
       expect(subject).to receive(:add_aio_defaults_on).exactly(hosts.length).times
       expect(subject).to receive(:add_puppet_paths_on).exactly(hosts.length).times
 
       subject.configure_foss_defaults_on( hosts )
     end
 
-    it 'uses foss paths for hosts of type foss' do
+    it 'uses no paths for hosts with no type' do
```

Comment:
I am interested in what happens when `:type` is set as `aio` because I know there are projects currently using this setting and I want to be sure something sane happens.